server/opts.go

```diff
@@ -615,6 +660,206 @@ func parseCluster(v interface{}, opts *Options, errors *[]error, warnings *[]err
 	return nil
 }
 
+func parseURLs(a []interface{}, typ string) ([]*url.URL, []error) {
+	var (
+		errors []error
+		urls   = make([]*url.URL, 0, len(a))
+	)
+	for _, u := range a {
+		tk, u := unwrapValue(u)
+		sURL := u.(string)
+		url, err := parseURL(sURL, typ)
+		if err != nil {
+			err := &configErr{tk, err.Error()}
+			errors = append(errors, err)
+			continue
+		}
+		urls = append(urls, url)
+	}
+	return urls, errors
+}
+
+func parseURL(u string, typ string) (*url.URL, error) {
+	urlStr := strings.TrimSpace(u)
+	url, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing %s url [%q]", typ, urlStr)
+	}
+	return url, nil
+}
+
+func parseGateway(v interface{}, o *Options, errors *[]error, warnings *[]error) error {
+	tk, v := unwrapValue(v)
+	gm, ok := v.(map[string]interface{})
+	if !ok {
+		return &configErr{tk, fmt.Sprintf("Expected gateway to be a map, got %T", v)}
+	}
+	for mk, mv := range gm {
+		// Again, unwrap token value if line check is required.
+		tk, mv = unwrapValue(mv)
+		switch strings.ToLower(mk) {
+		case "name":
```

Comment:
This is the other side's cluster name yes?

Reply:
Or our name, which should match clustername.

Reply:
There is no clustername for reason I described before. The config will look like this on a given server.
```
gateway {
  name: "A"
  port: 1234
  tls {
    ...
  }
  gateways [
       {
           name: "B"
           tls {
           }
           url: "nats://otherhost:1234
       }
   ]
}
```
Now if you tell me we require user to define cluster section to use gateways, then I could change things (quite a big change, but ok..).

Reply:
No I think it is ok as you have it defined.